packages/SwingSet/src/kernel/virtualObjectManager.js

```diff
@@ -222,6 +222,36 @@ export function makeVirtualObjectManager(
     return reachableVrefs.has(vref);
   }
 
+  /**
+   * Set of all import vrefs which are recognizable by our virtualized data.
+   * These were Presences at one point. We add to this set whenever we use a
+   * Presence as a key into a makeWeakStore() instance or an instance of
+   * VirtualObjectAwareWeakMap or VirtualObjectAwareWeakSet. We currently never
+   * remove anything from the set, but eventually we'll use refcounts to figure
+   * out when the vref becomes unrecognizable, allowing the vat do send a
+   * retireImport into the kernel.
+   *
+   */
+  /** @type {Set<string>} of vrefs */
+  const recognizableVrefs = new Set();
+
+  // XXX this function takes a value whereas its brethren (e.g.,
```

Comment:
Ah, right, because in the places that call this, they have a `vkey` but that's very much not the same thing as a `vref`.

Maybe we should refactor `virtualObjectKey` to take a `vref` instead? So the callers would do something like:
```js
const vref = getSlotForVal(key);
const vkey = virtualObjectKey(vref);
if (vkey) {
  addRecognizablePresence(vref);
  otherstuff(vkey);
  ..
```
?

Reply:
The whole point of `virtualObjectKey` is converting between an object reference and a key string with a vref as the intermediate state on the way there.  If it were to take a vref instead you'd just be adding some logic that would have to be replicated in a bunch of places.  The way it's factored now is right, I think, aside from the misleading parallel between the naming of `addReachablePresence` and `addRecognizablePresence`, so I think the proper approach is to address the naming directly.  I'm thinking `addReachablePresenceRef` and `addRecognizablePresenceValue`.